src/nunit.analyzers/BaseAssertionAnalyzer.cs

```diff
@@ -16,24 +16,24 @@ public override void Initialize(AnalysisContext context)
 
         protected abstract void AnalyzeAssertInvocation(OperationAnalysisContext context, IInvocationOperation assertOperation);
 
-        protected static bool IsAssert(IOperation operation)
+        protected bool IsAssert(IInvocationOperation invocationOperation)
         {
-            return operation is IExpressionStatementOperation expressionOperation &&
-                expressionOperation.Operation is IInvocationOperation invocationOperation &&
-                IsAssert(invocationOperation);
+            return this.IncludeClassic()
+                ? invocationOperation.TargetMethod.ContainingType.IsAnyAssert()
+                : invocationOperation.TargetMethod.ContainingType.IsAssert();
         }
 
-        protected static bool IsAssert(IInvocationOperation invocationOperation)
+        protected virtual bool IncludeClassic()
```

Comment:
Not the `virtual` method I meant. I would like `IsAssert` to become virtual, so there is no 'if' at all.
The base class implementation calls the `IsAssert` and the derived override calls `IsAnyAssert`

Reply:
Oh ok, got it